plugins/put-a-file-with-ftp/impl/src/test/java/org/pentaho/di/job/entries/ftpput/JobEntryFTPPUTTest.java

```diff
@@ -89,6 +114,43 @@ public void createFtpClient_UsesTimeoutIfSet() throws Exception {
     verify( ftpClient ).setTimeout( 10 );
   }
 
+  @Test
+  public void testLocalDirDoesntExist() throws Exception{
+    this.initialSetUp();
+    Result result = entry.execute( previousResult, 1 );
+    assertEquals( 1, result.getNrErrors() );
+  }
+
+  @Test
+  public void testNoFilePermissions() throws Exception {
+    this.initialSetUp();
+    Path path = Paths.get( "Localdir" );
+    try ( MockedStatic<Files> files = mockStatic( Files.class ) ) {
+      files.when(() -> Files.exists(any()))
+              .thenReturn(true);
+      Set<PosixFilePermission> permissions = Files.getPosixFilePermissions(path);
+      assertTrue(permissions.isEmpty());
+    }
+    Result result = entry.execute( previousResult, 1 );
+    assertEquals( 1, result.getNrErrors() );
+  }
+
+  public void initialSetUp() throws Exception {
+    JobEntryBase base = mock( JobEntryBase.class );
+    entry.setUserName( "username" );
+    entry.setPassword( "password" );
+    entry.setRemoteDirectory( "Remotedir" );
+    entry.setLocalDirectory( "Localdir" );
+    when( base.environmentSubstitute( "username" ) ).thenReturn( "mockedUsername" );
+    when( base.environmentSubstitute( "password" ) ).thenReturn( "mockedEncryptedPassword" );
+    try ( MockedStatic<Encr> encr = mockStatic( Encr.class ) ) {
+      encr.when( () -> Encr.decryptPasswordOptionallyEncrypted( anyString() ) )
```

Comment:
This is throwing an NullPointerException when I attach a debugger to it.

![Screenshot 2023-11-03 at 1 08 04 PM](https://github.com/pentaho/pentaho-kettle/assets/2119761/d3cec26a-430d-4006-98e3-3f0a5646c290)